source/priority_queue.c

```diff
@@ -176,6 +178,8 @@ void aws_priority_queue_init_static(
     size_t item_size,
     aws_priority_queue_compare_fn *pred) {
 
+    AWS_FATAL_PRECONDITION(queue);
```

Comment:
Why are the preconditions different? Why is one an ERROR and one a FATAL? Why does one do `!= NULL` but the other doesn't?

Reply:
The static methods are void so cannot return an error. I changed the dynamic inits to use FATAL  asserts instead of ERROR and modified the harnesses to explicitly make the same assumptions. This means we are not checking for the null ptrs directly anymore, but keeping the assumptions between the harness and the c code in agreement. 

In the case of priority_queue, some of the assumptions are redundant since it calls the array_list function but I thought it better to make everything explicit.